server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSHandleOptionsCheckUtility.java

```diff
@@ -0,0 +1,258 @@
+              put(CICSLexer.TIMERERR, ErrorSeverity.ERROR);
+              put(CICSLexer.SYMBOLERR, ErrorSeverity.ERROR);
+              put(CICSLexer.TEMPLATERR, ErrorSeverity.ERROR);
+              put(CICSLexer.NOTSUPERUSER, ErrorSeverity.ERROR);
+              put(CICSLexer.CSDERR, ErrorSeverity.ERROR);
+              put(CICSLexer.DUPRES, ErrorSeverity.ERROR);
+              put(CICSLexer.RESUNAVAIL, ErrorSeverity.ERROR);
+              put(CICSLexer.CHANNELERR, ErrorSeverity.ERROR);
+              put(CICSLexer.CCSIDERR, ErrorSeverity.ERROR);
+              put(CICSLexer.TIMEDOUT, ErrorSeverity.ERROR);
+              put(CICSLexer.CODEPAGEERR, ErrorSeverity.ERROR);
+              put(CICSLexer.INCOMPLETE, ErrorSeverity.ERROR);
+              put(CICSLexer.APPNOTFOUND, ErrorSeverity.ERROR);
+              put(CICSLexer.BUSY, ErrorSeverity.ERROR);
+            }
+          };
+
+  public CICSHandleOptionsCheckUtility(DialectProcessingContext context, List<SyntaxError> errors) {
+    super(context, errors, DUPLICATE_CHECK_OPTIONS);
+  }
+
+  /**
+   * Entrypoint to check CICS Handle rule options
+   *
+   * @param ctx ParserRuleContext subclass containing options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    switch (ctx.getRuleIndex()) {
+      case CICSParser.RULE_cics_handle_abend:
+        checkHandleAbend((CICSParser.Cics_handle_abendContext) ctx);
+        break;
+      case CICSParser.RULE_cics_handle_aid:
+        checkHandleAid((CICSParser.Cics_handle_aidContext) ctx);
+        break;
+      case CICSParser.RULE_cics_handle_condition:
+        checkHandleCondition((CICSParser.Cics_handle_conditionContext) ctx);
+        break;
+      default:
+        break;
+    }
+    checkDuplicates(ctx);
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkHandleAbend(CICSParser.Cics_handle_abendContext ctx) {
+    checkHasMandatoryOptions(ctx.ABEND(), ctx, "ABEND");
+    checkHasMutuallyExclusiveOptions("CANCEL or PROGRAM or LABEL or RESET", ctx.CANCEL(), ctx.PROGRAM(), ctx.LABEL(), ctx.RESET());
+  }
+
+  private void checkHandleAid(CICSParser.Cics_handle_aidContext ctx) {
+    checkHasMandatoryOptions(ctx.AID(), ctx, "AID");
+    checkHasTooManyOptions(ctx);
+  }
+
+  private void checkHandleCondition(CICSParser.Cics_handle_conditionContext ctx) {
+    checkHasMandatoryOptions(ctx.CONDITION(), ctx, "CONDITION");
+  }
+
+  private void checkHasTooManyOptions(ParserRuleContext parentCtx) {
+    List<ParseTree> commandOoptions = parentCtx.children.stream()
+            .filter(node -> node instanceof TerminalNode)
+            .filter(node -> !node.getText().equalsIgnoreCase("AID"))
+            .collect(Collectors.toList());
+    if (commandOoptions.size() > 16) {
+      throwException(
+              ErrorSeverity.ERROR, getLocality(parentCtx), "Too many options provided for: ", "HANDLE AID");
+    }
```

Comment:
The imports must be cleaned up...
```suggestion
    if (parentCtx.children == null)
      return;
    long commandOptionsCount = parentCtx.children.stream()
            .filter(node -> node instanceof TerminalNode)
            .map(TerminalNode.class::cast)
            .filter(node -> node.getSymbol().getType() != CICSLexer.AID)
            .count();
    if (commandOptionsCount > 16) {
      throwException(ErrorSeverity.ERROR, getLocality(parentCtx), "Too many options provided for: ", "HANDLE AID");
    }
```